_docs/info/migrate_entire_pool.md

```diff
@@ -0,0 +1,266 @@
+NAME                  USED  AVAIL  REFER  MOUNTPOINT
+tank1                1.58G  13.4G    24K  /tank1
+tank1/nfs             879M  13.4G   879M  /nfs
+tank1/vm-100-disk-0   734M  13.4G   734M  -
+```
+
+When all this is done it is now time to reboot into single user mode.
+
+After reboot into single user mode we start with creating a snapshot of the entire pool we want to migrate.
+
+``` terminal
+local-host $ sudo zfs snapshot -r tank1@replication
+
+local-host $ zfs list -r -t snapshot tank1
+NAME                              USED  AVAIL  REFER  MOUNTPOINT
+tank1@replication                    0      -    24K  -
+tank1/nfs@replication                0      -   879M  -
+tank1/vm-100-disk-0@replication      0      -   734M  -
+```
+
+After making the a snapshot of the entire pool when temporarily send the entire pool contents to another pool. The pool needs to be configured prior to sending the data and it needs to have a size at least equal to the size of the pool we migrate.
+
+**Local operation**
+
+``` terminal
+local-host $ sudo zfs send -R tank1@replication | sudo zfs recv -Fdu tank2
+
+local-host $ zfs list -r tank2
+NAME                  USED  AVAIL  REFER  MOUNTPOINT
+tank2                1.58G  13.4G    24K  /tank2
+tank2/nfs             879M  13.4G   879M  /nfs
+tank2/vm-100-disk-0   734M  13.4G   734M  -
+
+local-host $ zfs list -r -t snapshot tank2
+NAME                              USED  AVAIL  REFER  MOUNTPOINT
+tank2@replication                    0      -    24K  -
+tank2/nfs@replication                0      -   879M  -
+tank2/vm-100-disk-0@replication      0      -   734M  - z
+```
+
+**Remote operation**
+
+``` terminal
+local-host $ sudo zfs send -R tank1@replication | remote-host $ sudo zfs recv -Fdu tank2
+
+remote-host $ zfs list -r tank2
+NAME                  USED  AVAIL  REFER  MOUNTPOINT
+tank2                1.58G  13.4G    24K  /tank2
+tank2/nfs             879M  13.4G   879M  /nfs
+tank2/vm-100-disk-0   734M  13.4G   734M  -
+
+remote-host $ zfs list -r -t snapshot tank2
+NAME                              USED  AVAIL  REFER  MOUNTPOINT
+tank2@replication                    0      -    24K  -
+tank2/nfs@replication                0      -   879M  -
+tank2/vm-100-disk-0@replication      0      -   734M  - z
+```
+
+After we have insured our destination pool contains the same as our source pool we can safely destroy the replication snapshots on both source and destination pools as well as destroying the old source pool and recreate a new source pool. As stated at the beginning if you want to use another name for the new source pool at extra operation is needed which is shown at the end.
+
+**Local operation**
+
+``` terminal
+local-host $ sudo zfs destroy -r tank1@replication
+local-host $ sudo zfs destroy -r tank2@replication
+
+local-host $ sudo zpool destroy -f tank1
+
+local-host $ sudo zpool create tank1 c1t3d0
+```
+
+**Remote operation**
+
+``` terminal
+local-host $ sudo zfs destroy -r tank1@replication
```

Comment:
local or remote ?

Reply:
If temporary pool is remote most steps will be a mix of local and remote commands so operations means what have to be done if the temporary pool is local or remote.

I could revert it to only one step instruction for each step and then do something like:
```
if local temporary pool
     .....
else
     .....
```

Reply:
ok ... maybe it would make sense to have two chapters ... one for localhost version and a second one for remote host version instead of mixing the two ... since someone following your recipe will only be doing one thing at a time ... either local or remote

Reply:
Should I split it up into two articles?

Reply:
No, I think one article is good, but maybe open up with some general explanation on the concept what you are about to do and then have two recipes one for the local case and one for the remote case ... in that way users can follow the right one.

Reply:
See last commit. Hope the methods, and the distinction between them,  is more clear now?